File: src/community/css/src/main/scala/org/geoserver/community/css/web/SummaryTable.scala
package org.geoserver.community.css.web

import java.io.Serializable

import scala.collection.JavaConverters._

import org.apache.wicket.Component
import org.apache.wicket.markup.html.basic.Label
import org.apache.wicket.model.Model
import org.apache.wicket.model.IModel

import org.geoserver.web.wicket.GeoServerDataProvider.Property
import org.geoserver.web.wicket.GeoServerTablePanel

/**
 * A specialized GeoServerTablePanel that shows maximum and minimum data for a
 * particular featuretype.  It is intended to provide some extra context during
 * styling, etc.
 *
 * @author David Winslow <cdwinslow@gmail.com>
 */
class SummaryTable(id: String, summary: SummaryProvider)
extends GeoServerTablePanel[Summary](id, summary) {
  override def getComponentForProperty(
    id: String,
    value: IModel[_],
    property: Property[Summary]
  ): Component =
    new Label(id, new Model(property.getPropertyValue(value.getObject.asInstanceOf[Summary]).toString))
}

File: src/community/css/src/main/java/org/geoserver/community/css/web/SummaryTable.java
/* Copyright (c) 2012 TOPP - www.openplans.org. All rights reserved.
 * This code is licensed under the GPL 2.0 license, available at the root
 * application directory.
 */
package org.geoserver.community.css.web;

import org.apache.wicket.Component;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.model.IModel;
import org.geoserver.web.wicket.GeoServerTablePanel;
import static org.geoserver.web.wicket.GeoServerDataProvider.Property;

public class SummaryTable extends GeoServerTablePanel<Summary> {
    public SummaryTable(String id, SummaryProvider summary) {
        super(id, summary);
    }

    @Override 
    protected Component getComponentForProperty(String id, IModel value, Property<Summary> property) {
        String text = property.getPropertyValue((Summary)value.getObject()).toString();
        return new Label(id, text);
    }
}
